src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionContributor.kt

```diff
@@ -238,6 +240,10 @@ class RsKeywordCompletionContributor : CompletionContributor(), DumbAware {
         return psiElement(IDENTIFIER).withParent(parent)
     }
 
+    private fun traitOrImplPattern(): PsiElementPattern.Capture<PsiElement> {
+        return psiElement().inside(RsTraitOrImpl::class.java).and(statementBeginningPattern())
```

Comment:
`inside` iterates over all parents of psi element. It leads to:
* pattern accepts more places than desired
* pattern make much more work than really needed
Let's make a more precise pattern. Looks like we can just check that our parent is `RsMembers`